### PR TITLE
docs: clarify behavior of flag initializer kwargs with aliases

### LINKED ISSUE #607: flags: the behavior when creating or updating flags with both an alias and its actual flag provided are not explained in the documentation
### Summary

Document that setting both an alias and a default value in flag creation will set the alias before setting the default value.

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Resolves #607.

The description in that issue isn't fully correct, essentially it boils down to "later kwargs take precedence over earlier kwargs".

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: changelog/749.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Clarify behavior of flag type initializers when both a flag and an alias are given.
```
Comment (onerandomusername) [inline changelog/749.doc.rst:1]: also affects update methods.

File: disnake/flags.py [hunk 2]
```diff
@@ -829,6 +829,7 @@ class Intents(BaseFlags):
 
     To construct an object you can pass keyword arguments denoting the flags
     to enable or disable.
+    Arguments are applied in order, similar to :class:`Permissions`.
```
Comment (onerandomusername) [inline disnake/flags.py:832]: this should also be under the other times that 
> `To construct an object you can pass keyword arguments denoting the flags` 

is mentioned

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/flags.py (hunk 2 above)
Comment (shiftinv) [reply disnake/flags.py:833]: Those other types currently don't have any aliased attributes which is why I didn't add it initially, but fine by me
https://github.com/DisnakeDev/disnake/pull/749/commits/9d3c82903aedbe5b4226a8c6259ab1c57e0d609b

File: changelog/749.doc.rst (hunk 1 above)
Comment (shiftinv) [reply changelog/749.doc.rst]: https://github.com/DisnakeDev/disnake/pull/749/commits/b1a41f0ab43233e4cf31ad3e8d6f836daf3b554f

**Review by EQUENOS — APPROVED** [review]

**Comment by onerandomusername** [comment]
Could we also test an alias with multiple bits, like Intents.messages or Intents.reactions?

**Comment by shiftinv** [comment]
> Could we also test an alias with multiple bits, like Intents.messages or Intents.reactions?

Those are already covered by `TestFlags.three`, which is `TestFlags.one | TestFlags.two`.

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**